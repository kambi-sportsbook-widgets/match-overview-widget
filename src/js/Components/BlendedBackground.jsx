import React, { PropTypes } from 'react';
import styles from './BlendedBackground.scss';

const BlendedBackground = () => (
   <svg xmlns='http://www.w3.org/2000/svg' xmlnsXlink='http://www.w3.org/1999/xlink' className={styles.background}>
      <defs>
         <filter id='filter'>
            <feImage
               id='background-image' /* @todo: replace with ref */
               xlinkHref='src/custom/overview-bw-bg-desktop.jpg'
               result='slide2'
               x='0'
               y='0'
               width='100%'
               preserveAspectRatio='xMidYMid slice'
            />
            <feBlend in2='SourceGraphic' in='slide2' mode='multiply' />
         </filter>
      </defs>
      <rect id='blendRect' x='0' y='0' filter='url(#filter)' width='100%' height='100%' />
   </svg>
);

BlendedBackground.propTypes = {
};

export default BlendedBackground;
