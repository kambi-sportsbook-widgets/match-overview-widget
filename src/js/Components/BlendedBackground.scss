:local(.backgroundContainer) {
   filter: brightness(90%);
}

:local(.background) {
   position: absolute;
   top: 0;
   right: 0;
   left: 0;
   bottom: 0;
   display: block;
   width: 100%;
   min-width: 100%;
   min-height: 100%;
   overflow: hidden;
}

:local(.blendRect) {
   fill: currentColor;
}
