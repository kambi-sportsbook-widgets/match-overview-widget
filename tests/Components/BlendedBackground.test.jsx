/* eslint-env jest */
import React, { Children } from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import ReactTestRenderer from 'react-test-renderer';
import { mount } from 'enzyme';
import BlendedBackground from '../../src/js/Components/BlendedBackground';

let renderer;
let mockMobile = false;

jest.mock('../../src/js/Services/mobile', () => () => mockMobile);
jest.mock('../../src/custom/overview-bw-bg-desktop.jpg', () => 'overview-bw-bg-desktop.jpg');
jest.mock('../../src/custom/overview-bw-bg-mobile.jpg', () => 'overview-bw-bg-mobile.jpg');

describe('BlendedBackground DOM rendering', () => {

   beforeEach(() => {
      renderer = new ReactShallowRenderer();
      mockMobile = false;
   });

   it('renders correctly with default props', () => {
      expect(renderer.render(
         <BlendedBackground />
      )).toMatchSnapshot();
   });

   it('renders correctly on mobile device', () => {
      mockMobile = true;

      expect(renderer.render(
         <BlendedBackground />
      )).toMatchSnapshot();
   });

   it('resizes correctly to mobile mode', () => {
      mockMobile = false;

      const tree = ReactTestRenderer.create(
         <BlendedBackground />
      );

      expect(tree).toMatchSnapshot();

      mockMobile = true;

      window.dispatchEvent(new Event('resize'));

      expect(tree).toMatchSnapshot();
   });

   it('resizes correctly to desktop mode', () => {
      mockMobile = true;

      const tree = ReactTestRenderer.create(
         <BlendedBackground />
      );

      expect(tree).toMatchSnapshot();

      mockMobile = false;

      window.dispatchEvent(new Event('resize'));

      expect(tree).toMatchSnapshot();
   });

   it('stays in proper mode on non-significant screen size change', () => {
      const tree = ReactTestRenderer.create(
         <BlendedBackground />
      );

      expect(tree).toMatchSnapshot();

      window.dispatchEvent(new Event('resize'));

      expect(tree).toMatchSnapshot();
   });

});

describe('BlendedBackground interface', () => {

   beforeEach(() => {
      renderer = new ReactShallowRenderer();
   });

   it('unmounts correctly', () => {
      const wrapper = mount(
         <BlendedBackground />
      );
      wrapper.unmount();
   });

});
