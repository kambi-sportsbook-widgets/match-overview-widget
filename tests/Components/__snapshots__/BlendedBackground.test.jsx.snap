// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlendedBackground DOM rendering renders correctly on mobile device 1`] = `
<div
  className={undefined}
>
  <svg
    className={undefined}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <filter
        id="filter"
      >
        <feImage
          preserveAspectRatio="xMidYMid slice"
          result="slide2"
          width="100%"
          x="0"
          xlinkHref="assets/overview-bw-bg-mobile.jpg"
          y="0"
        />
        <feBlend
          in="slide2"
          in2="SourceGraphic"
          mode="multiply"
        />
      </filter>
    </defs>
    <rect
      className="KambiWidget-primary-color undefined"
      filter="url(#filter)"
      height="100%"
      width="100%"
      x="0"
      y="0"
    />
  </svg>
</div>
`;

exports[`BlendedBackground DOM rendering renders correctly with default props 1`] = `
<div
  className={undefined}
>
  <svg
    className={undefined}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <filter
        id="filter"
      >
        <feImage
          preserveAspectRatio="xMidYMid slice"
          result="slide2"
          width="100%"
          x="0"
          xlinkHref="assets/overview-bw-bg-desktop.jpg"
          y="0"
        />
        <feBlend
          in="slide2"
          in2="SourceGraphic"
          mode="multiply"
        />
      </filter>
    </defs>
    <rect
      className="KambiWidget-primary-color undefined"
      filter="url(#filter)"
      height="100%"
      width="100%"
      x="0"
      y="0"
    />
  </svg>
</div>
`;

exports[`BlendedBackground DOM rendering resizes correctly to desktop mode 1`] = `
<div
  className={undefined}
>
  <svg
    className={undefined}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <filter
        id="filter"
      >
        <feImage
          preserveAspectRatio="xMidYMid slice"
          result="slide2"
          width="100%"
          x="0"
          xlinkHref="assets/overview-bw-bg-mobile.jpg"
          y="0"
        />
        <feBlend
          in="slide2"
          in2="SourceGraphic"
          mode="multiply"
        />
      </filter>
    </defs>
    <rect
      className="KambiWidget-primary-color undefined"
      filter="url(#filter)"
      height="100%"
      width="100%"
      x="0"
      y="0"
    />
  </svg>
</div>
`;

exports[`BlendedBackground DOM rendering resizes correctly to desktop mode 2`] = `
<div
  className={undefined}
>
  <svg
    className={undefined}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <filter
        id="filter"
      >
        <feImage
          preserveAspectRatio="xMidYMid slice"
          result="slide2"
          width="100%"
          x="0"
          xlinkHref="assets/overview-bw-bg-desktop.jpg"
          y="0"
        />
        <feBlend
          in="slide2"
          in2="SourceGraphic"
          mode="multiply"
        />
      </filter>
    </defs>
    <rect
      className="KambiWidget-primary-color undefined"
      filter="url(#filter)"
      height="100%"
      width="100%"
      x="0"
      y="0"
    />
  </svg>
</div>
`;

exports[`BlendedBackground DOM rendering resizes correctly to mobile mode 1`] = `
<div
  className={undefined}
>
  <svg
    className={undefined}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <filter
        id="filter"
      >
        <feImage
          preserveAspectRatio="xMidYMid slice"
          result="slide2"
          width="100%"
          x="0"
          xlinkHref="assets/overview-bw-bg-desktop.jpg"
          y="0"
        />
        <feBlend
          in="slide2"
          in2="SourceGraphic"
          mode="multiply"
        />
      </filter>
    </defs>
    <rect
      className="KambiWidget-primary-color undefined"
      filter="url(#filter)"
      height="100%"
      width="100%"
      x="0"
      y="0"
    />
  </svg>
</div>
`;

exports[`BlendedBackground DOM rendering resizes correctly to mobile mode 2`] = `
<div
  className={undefined}
>
  <svg
    className={undefined}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <filter
        id="filter"
      >
        <feImage
          preserveAspectRatio="xMidYMid slice"
          result="slide2"
          width="100%"
          x="0"
          xlinkHref="assets/overview-bw-bg-mobile.jpg"
          y="0"
        />
        <feBlend
          in="slide2"
          in2="SourceGraphic"
          mode="multiply"
        />
      </filter>
    </defs>
    <rect
      className="KambiWidget-primary-color undefined"
      filter="url(#filter)"
      height="100%"
      width="100%"
      x="0"
      y="0"
    />
  </svg>
</div>
`;

exports[`BlendedBackground DOM rendering stays in proper mode on non-significant screen size change 1`] = `
<div
  className={undefined}
>
  <svg
    className={undefined}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <filter
        id="filter"
      >
        <feImage
          preserveAspectRatio="xMidYMid slice"
          result="slide2"
          width="100%"
          x="0"
          xlinkHref="assets/overview-bw-bg-desktop.jpg"
          y="0"
        />
        <feBlend
          in="slide2"
          in2="SourceGraphic"
          mode="multiply"
        />
      </filter>
    </defs>
    <rect
      className="KambiWidget-primary-color undefined"
      filter="url(#filter)"
      height="100%"
      width="100%"
      x="0"
      y="0"
    />
  </svg>
</div>
`;

exports[`BlendedBackground DOM rendering stays in proper mode on non-significant screen size change 2`] = `
<div
  className={undefined}
>
  <svg
    className={undefined}
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <filter
        id="filter"
      >
        <feImage
          preserveAspectRatio="xMidYMid slice"
          result="slide2"
          width="100%"
          x="0"
          xlinkHref="assets/overview-bw-bg-desktop.jpg"
          y="0"
        />
        <feBlend
          in="slide2"
          in2="SourceGraphic"
          mode="multiply"
        />
      </filter>
    </defs>
    <rect
      className="KambiWidget-primary-color undefined"
      filter="url(#filter)"
      height="100%"
      width="100%"
      x="0"
      y="0"
    />
  </svg>
</div>
`;
